src/main/java/org/magicdgs/thaplv/haplotypes/model/AbstractHaplotypeConverter.java

```diff
@@ -0,0 +1,152 @@
+     */
+    public Genotype apply(final Genotype genotype) {
+        // early termination for already haploid individuals
+        if (genotype.getAlleles().size() == 1) {
+            return (getOutputPlotidy() == 1) ? genotype :
+                    new GenotypeBuilder(genotype).phased(true)
+                            .alleles(Collections.nCopies(getOutputPlotidy(), genotype.getAllele(0)))
+                            .make();
+        }
+        final GenotypeBuilder builder = new GenotypeBuilder(genotype).phased(true);
+        if (isConsideredMissing(genotype)) {
+            return builder.alleles(missingGenotype).make();
+        } else {
+            return builder.alleles(Collections.nCopies(missingGenotype.size(),
+                    getHaplotypeAllele(genotype.getAlleles())))
+                    .make();
+        }
+    }
+
+    /**
+     * The output ploidy is the length of {@link #missingGenotype}
+     */
+    public int getOutputPlotidy() {
+        return missingGenotype.size();
+    }
+
+    /**
+     * Convert a list of alleles from a single individual into the alleles for the haplotype.
+     *
+     * WARNING: {@link AbstractHaplotypeConverter} only consider genotypes with 2 alleles or less.
+     *
+     * @param alleles the alleles from the individual
+     *
+     * @return the allele in the haplotype from the individual
+     */
+    protected Allele getHaplotypeAllele(final List<Allele> alleles) {
+        // converting the allele to a set to quick check if they are not equals
+        final Set<Allele> alleleSet = new HashSet<>(alleles);
+        // the allele to return
+        final Allele toReturn;
+        switch (alleleSet.size()) {
+            // if the set is 1, return the allele
+            case 1:
+                toReturn = alleles.get(0);
+                break;
+            case 2:
+                toReturn = getHaplotypeAllele(alleles.get(0), alleles.get(1));
+                break;
+            default:
+                throw new IllegalArgumentException(
+                        "This software cannot handle genotypes with more than 2 alleles");
+        }
+        return toReturn;
+    }
+
+    /**
+     * Is this genotype considered missing by this model? It should return always true if the
+     * genotype is completely missing.
+     *
+     * Behaviour of this method may be unexpected for non-diploid individuals in implementations.
+     *
+     * @param genotype the genotype to test
+     *
+     * @return {@code true} if it is considered missing; {@code false} otherwise
+     */
+    protected abstract boolean isConsideredMissing(final Genotype genotype);
+
+    /**
+     * Get the haplotype for two alleles that are different and are not considered missing by this
+     * HaplotypeConverter.
+     *
+     * Behaviour may be unexpected for alleles which do not pass the {@link
+     * #isConsideredMissing(Genotype)} filter in implementations.
```

Comment:
Rephrase: "Implementations don't need to handle situations where  alleles do not pass the isConsideredMissing"